### PR TITLE
[CCP] make timeframes configurable

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
This pull request addresses this [task](https://jira.mitre.org/browse/ASCODCP-2058).  The timescales being used by the compass app were hardcoded as 1,3, and 5 years.  Now they can be configured by altering the state in `TreatmentOptionsOutcomes`.  The filtering of patients based on the number of years they survived has been passed from the treatment filter function to the components that actually need that information.  

Instead of receiving an object with the 1yr, 3yr, and 5yr survival rates already calculated, the components now receive an array where each index corresponds to # of years of survival.  The components use this information to calculate the survival rate for any arbitrary year value.

Currently the only way to change the year intervals is through the code in the state of `TreatmentOptionsOutcomes`.  Some sort of ui component to change these values could be implemented in another task.  Though I could add some UI component for this PR as well, if the task description isn't met by what this PR does.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by Dtphelan1** [comment]
[DP1] - SurvivalYears seems to actually be an array of the deathsPerYear. Let's change the name to reflect that fact and avoid confusion. 
[DP2] - Instead of defining our own Sum function, let's just use the one provided by [Lodash](https://lodash.com/docs/4.17.11#sum), slicing our array as needed before passing as input. 
[DP3] - In RenderHeader we map over timeScales to make sure we render a header cell for each element. In RenderRow we render elements 0,1,2 of timeScales, hardcoded as such. Can we map over timescales in RenderRow as we do in RenderHeader?
[DP4] - The actual definition of done for this task describes using a `timeframes` parameter, but I don't see anything named like this. Could we speak on this a bit? Also - do we have anywhere in our code where we make a mock-call out to the service? If so, have we updated that mock-call to accept either this `timeframes` parameter or something like that? 

And a general question about the approach - this assumes that year is the minimal level of granularity. Is that sufficient to support what is expected of Compass and of this task? Paging @rdingwell on this.

File: src/__test__/backend/mcode/mock-data/index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const similarPatientTreatmentsData = [
         "id": "row_2",
         "displayName": "hormonal therapy",
         "totalPatients": 12,
-        "survivalYears":[2,0,2,0,6,0],
+        "deathsPerYear":[2,0,2,0,6,0],
```
Comment (rdingwell) [inline src/__test__/backend/mcode/mock-data/index.js]: I think I may have missed this in previous review.  But is this stating that what has been the number of patients who have survived in the past has now been replaced by the number who have died?  If so what was the reasoning behind this change?  I'm not real sure about it.

File: src/reducers/mcode.js [hunk 2]
```diff
@@ -18,11 +19,13 @@ export default function mcode(state = defaultState, action) {
             isSimilarPatient(treatmentDataPatient, state.similarPatientProps));
         const similarPatientTreatments = generateSimilarPatientTreatments(similarPatients);
 
+        const timescale = ['1','3','5'];
```
Comment (rdingwell) [inline src/reducers/mcode.js]: However we decide that the timescales will be configurable they must be a property that can be configured in the config.js file that is located in the public folder.

File: src/__test__/backend/mcode/mock-data/index.js (hunk 1 above)
Comment (Dtphelan1) [reply src/__test__/backend/mcode/mock-data/index.js]: Based on my assessment of how this data gets used, yes that's how this array gets used.

**Comment by Dtphelan1** [comment]
Should we schedule a 15 minute meeting to get on the same page  re this task? I feel like a bunch of open questiosn have been posed and there hasn't been any discussion about them here. @rdingwell  and @KeeyanGhoreshi ?

File: src/__test__/backend/mcode/mock-data/index.js [hunk 3]
```diff
@@ -66,9 +62,7 @@ export const similarPatientTreatmentsData = [
         "id": "row_4",
         "displayName": "test therapy",
         "totalPatients": 10,
-        "oneYrSurvival": 9,
-        "threeYrSurvival": 9,
-        "fiveYrSurvival": 8,
+        "survivorsPerYear":[10,9,9,9,10,10],
```
[the remark this answers is not in the document]
Comment (KeeyanGhoreshi) [reply src/__test__/backend/mcode/mock-data/index.js]: No, it's just to test sorting really.  I didn't want to change the other treatment numbers, so I just made this test treatment to ensure order changed when sorted by different years.

[the remark this answers is not in the document]
Comment (noranda) [reply src/__test__/backend/mcode/mock-data/index.js]: Understood, although we may want to test other things using this property in the future, so it probably makes sense for it to reflect data that's actually possible.

**Comment by noranda** [comment]
[NB1] resolved. Once @Dtphelan1 and @rdingwell confirm their issues are resolved and the branch is rebased, I will approve. 👍

File: src/mcode-pilot/components/TreatmentOptionsOutcomes/TreatmentOptionsOutcomes.jsx [hunk 4]
```diff
@@ -91,13 +90,17 @@ export default class TreatmentOptionsOutcomes extends Component {
     }
 
     renderToggleButtons = () => {
-        const { outcomesToggle, timescaleToggle } = this.state;
-        const timeScales = [1, 3, 5];
+        let { outcomesToggle, timescaleToggle} = this.state;
```
Comment (Dtphelan1) [inline src/mcode-pilot/components/TreatmentOptionsOutcomes/TreatmentOptionsOutcomes.jsx]: Minor nitpick - add a space to separate the curly and Toggle

**Comment by Dtphelan1** [comment]
Other than the minor change I see here, I'm good to approve.

**Comment by KeeyanGhoreshi** [comment]
This branch should now be rebased.

**Comment by rdingwell** [comment]
Once this has been rebased and pushed I'm fine with it being merged into master.
